services/http/api/lib/models/hour.ts

```diff
@@ -0,0 +1,14 @@
+import { ModelRelatedNodesI, NeogmaInstance, NeogmaModel } from "neogma";
+import { MeasurementValueModel, UserMeasurementModel } from "./measurement";
+
+export type HourProperties = {
+    time: number
+}
+
+export type HourRelatedNodeI = {
+    Value: ModelRelatedNodesI<UserMeasurementModel, MeasurementValueModel>
+}
+
+export type HourInstance = NeogmaInstance<HourProperties, HourRelatedNodeI>;
+
+export type HourModel = NeogmaModel<HourProperties, HourRelatedNodeI>;
```

Comment:
I'm not fully satisfied with the date system yet. In particular, I think that hour should have a single property which refers to the hour in the day (i.e. we should only have 24 hour nodes in the entire database) and the specific time for a measurement should be stored with the measurement itself. Here are a couple of proposed schemas I came up with:

1. Incorporating the ideas above:
![Image](https://user-images.githubusercontent.com/4219194/122256893-53b04d00-ce84-11eb-9128-a5ac938aff6d.jpeg)

2. Using more structure for the date representation (this results in fewer node instances so better performance in the db)
![Image 2](https://user-images.githubusercontent.com/4219194/122257055-7cd0dd80-ce84-11eb-90f3-a4ba310e5770.jpeg)
Note that the includes relationship is needed from month to day to tell whether or not a month has that day in it (e.g. December-Includes->31, but !February-Includes->31). Obviously, the primary key for each year, day,month, hour node will be the num field, which eliminates the need for the string generation currently used.

Reply:
I'm good with either design but per Aidin's comments I think that we like more node structure and I like that the second design has fewer date node instances. also allows more efficient queries for year, month, day.

Reply:
Aside from day being included in month, this has already been implemented in qol-survey branch, which has been merged into here. I'm going to resolve this.